### PR TITLE
Fix dgfs

### PR DESCRIPTION
**Pull request by teddygroves**
This change partially implements #442, making it possible to fix the values of formation energy parameters in the parameter input file.

The interface is as follows for independent dgf input:

```toml
dgf = [
  {location = -1, metabolite = "M1", scale = 0.05, fixed_value = -1},
]
```

or like this for multivariate input:

```toml
[dgf]
ids = ["M1", "M2", "M3"]
mean_vector = [1, 2, 3]
covariance_matrix = [[1,2,3], [1, 2, 3], [1, 2, 3]]
[dgf.fixed_values]
M1 = -1
```

I think this change is just about ready~~, the only thing to do is update the documentation~~.

Checklist:

- [x] Updated any relevant documentation
- [ ] Add an adr doc if appropriate
- [x] Include links to any relevant issues in the description
- [x] Unit tests passing
- [x] Integration tests passing

### CODE REVIEW

**Review by carrascomj — CHANGES_REQUESTED** [review]
I am getting and error where no `[dgf.fixed_values]` block is specified in the priors.

````python
Traceback (most recent call last):
  File "/home/georg/.virtualenvs/maud-branch/bin/maud", line 8, in <module>
    sys.exit(cli())
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/cli.py", line 166, in simulate_command
    click.echo(do_simulate(data_path, output_dir, n))
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/cli.py", line 193, in do_simulate
    stanfit = simulate(mi, samples_path, n)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/running_stan.py", line 152, in simulate
    set_up_output_dir(output_dir, mi)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/running_stan.py", line 191, in set_up_output_dir
    cmdstanpy.utils.write_stan_json(input_path_train, mi.stan_input_train)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/pydantic/main.py", line 756, in __getattr__
    return super().__getattribute__(item)  # Raises AttributeError if appropriate
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/data_model/maud_input.py", line 41, in stan_input_train
    train, _ = get_stan_inputs(
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/getting_stan_inputs.py", line 45, in get_stan_inputs
    fixed_param_input = get_fixed_param_input(parameters, kinetic_model)
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/getting_stan_inputs.py", line 79, in get_fixed_param_input
    "ix_dgf_free": list(range(1, len(dgf.location) + 1)),
  File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/pydantic/main.py", line 759, in __getattr__
    raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
AttributeError: 'Dgf' object has no attribute 'location'
```

**Comment by teddygroves** [comment]
> I am getting and error where no `[dgf.fixed_values]` block is specified in the priors.
> 
> ```python
> Traceback (most recent call last):
>   File "/home/georg/.virtualenvs/maud-branch/bin/maud", line 8, in <module>
>     sys.exit(cli())
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
>     return self.main(*args, **kwargs)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1078, in main
>     rv = self.invoke(ctx)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
>     return _process_result(sub_ctx.command.invoke(sub_ctx))
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
>     return ctx.invoke(self.callback, **ctx.params)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/click/core.py", line 783, in invoke
>     return __callback(*args, **kwargs)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/cli.py", line 166, in simulate_command
>     click.echo(do_simulate(data_path, output_dir, n))
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/cli.py", line 193, in do_simulate
>     stanfit = simulate(mi, samples_path, n)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/running_stan.py", line 152, in simulate
>     set_up_output_dir(output_dir, mi)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/running_stan.py", line 191, in set_up_output_dir
>     cmdstanpy.utils.write_stan_json(input_path_train, mi.stan_input_train)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/pydantic/main.py", line 756, in __getattr__
>     return super().__getattribute__(item)  # Raises AttributeError if appropriate
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/data_model/maud_input.py", line 41, in stan_input_train
>     train, _ = get_stan_inputs(
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/getting_stan_inputs.py", line 45, in get_stan_inputs
>     fixed_param_input = get_fixed_param_input(parameters, kinetic_model)
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/maud/getting_stan_inputs.py", line 79, in get_fixed_param_input
>     "ix_dgf_free": list(range(1, len(dgf.location) + 1)),
>   File "/home/georg/.virtualenvs/maud-branch/lib/python3.10/site-packages/pydantic/main.py", line 759, in __getattr__
>     raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
> AttributeError: 'Dgf' object has no attribute 'location'
> ```
> ```

Thanks - I think this is fixed by [ba18f98](https://github.com/biosustain/Maud/pull/449/commits/ba18f988bf81c700ecd3f4c6f8ff5cb368bb25a0)

**Review by carrascomj — APPROVED** [review]
It needs a rebase, otherwise LGTM!

File: maud/data_model/maud_parameter.py [hunk 1]
```diff
@@ -33,6 +33,51 @@ class MaudParameter(BaseModel):
     split_ids: List[List[List[str]]]
     measurements: Optional[List[Measurement]] = None
 
+    @computed_field
+    def fixed_ids(self) -> Optional[List[List[str]]]:
+        """Set the fixed_ids field."""
+        if self.name != "dgf":
```
[the remark this answers is not in the document]
Comment (teddygroves) [reply maud/data_model/maud_parameter.py:39]: Yeah there's definitely lots of room for improvement on how this is implemented, especially the inits and stan code generation. I think it'll be nice to think about this between now and when we want to generalise to other (all?) parameters.

**Pull request merged**